components/header.js

```diff
@@ -1,16 +1,199 @@
-import React from 'react';
+'use client';
+import React, { useState } from 'react';
 import pageStyles from '../app/page.module.css';
+import { icons } from '@/components/utilities/card';
+import NavHeader from './utilities/navHeader';
 
 const Header = () => {
+  const [headerRules, setHeaderRules] = useState({ cardRank: false, suitRank: false, combinations: false, smashes: false, gameplay: false });
+  const [toggleRules, setToggleRules] = useState(false);
+
+  const handleClick = (type) => {
+    setHeaderRules({
+      ...headerRules,
+      [type]: !headerRules[type],
+    });
+  }
+
   return (
     <header className={pageStyles.header}>
       <h1>Thirteen</h1>
       <nav>
         <ul>
-          <li><button>Rules</button></li>
+          <li><button onClick={() => setToggleRules(true)}>Rules</button></li>
           <li><button>Dark Mode</button></li>
         </ul>
       </nav>
+      <div className={toggleRules ? pageStyles.rules : pageStyles.hide}>
+        <div className={pageStyles.rulesHeader}>
+          <h2>Rules</h2>
+          <button onClick={() => setToggleRules(false)}>
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><path d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"/></svg>
+          </button>
+        </div>
+        <div className={pageStyles.cardRanks}>
+          <NavHeader handleClick={handleClick} rules={headerRules} type={'cardRank'} />
```

Comment:
it's a better UX to make the whole div clickable as opposed to just the caret being clickable